File: src/app/Models/order.ts
import { Address } from "./address";
import { Payment } from "./payment";

export class Order {
    _id: string; 
    customerId: string; 
    // sellerId: string; 
    productId: string;
    orderedDate: Date | undefined; 
    orderQuantity: number;
    deliveredDate: Date | undefined;
    status: string; 
    shippingAddress: Address;
    paymentDetails: Payment; 
    totalAmount: number; 

    constructor(
        _id: string, 
        customerId: string, 
        // sellerId: string, 
        productId: string, 
        orderedDate: Date | undefined,
        orderQuantity: number,
        deliveredDate: Date | undefined,
        status: string, 
        shippingAddress: Address,
        paymentDetails: Payment, 
        totalAmount: number
    ) {
        this._id = _id;
        this.customerId = customerId;  
        // this.sellerId =  sellerId;
        this.productId = productId;
        this.orderedDate = orderedDate;
        this.orderQuantity = orderQuantity;
        this.deliveredDate = deliveredDate;
        this.status = status;
        this.shippingAddress = shippingAddress; 
        this.paymentDetails = paymentDetails;
        this.totalAmount = totalAmount;
    }   
}

File: GoServer/Model/order.go
package Model

type Order struct {
	Id        			string      `json:"_id" bson:"_id"`
	CustomerId        	string      `json:"customerId" bson:"customerId"`
	// SellerId 			string 		`json:"sellerId" bson:"sellerId"`	
	ProductId        	string      `json:"productId" bson:"productId"`
	OrderQuantity		int			`json:"orderQuantity" bson:"orderQuantity"`
	OrderedDate			string   	`json:"orderedDate" bson:"orderedDate"`
	DeliveredDate		string   	`json:"deliveredDate" bson:"deliveredDate"`
	Status		    	string   	`json:"status" bson:"status"`
	ShippingAddress    	Address   	`json:"shippingAddress" bson:"shippingAddress"`
	PaymentDetails		Payment   	`json:"paymentDetails" bson:"paymentDetails"` 
	TotalAmount 		int 		`json:"totalAmount" bson:"totalAmount"`
}